common/chat.ts

```diff
@@ -15,10 +15,20 @@ export enum ClientChatMessageType {
 
 export type ChatMessageType = ServerChatMessageType | ClientChatMessageType
 
-// TODO(2Pac): Include more information here, e.g. channel permissions, join date, etc.
+export interface ChannelPermissions {
+  kick: boolean
+  ban: boolean
+  changeTopic: boolean
+  togglePrivate: boolean
+  editPermissions: boolean
+  owner: boolean
+}
+
 export interface ChatUser {
   id: SbUserId
   name: string
+  joinDate: number
+  permissions: ChannelPermissions
```

Comment:
I do not think it makes sense to send channel permissions about every user to every other user in chat, that seems like a massive information leak to me. I understand you may need this information for some UIs, but those UIs should retrieve it as they need it (and it should be limited to users that should have permission to do so)